src/dbnode/persist/fs/commitlog/chunk_reader.go

```diff
@@ -80,18 +83,31 @@ func (r *chunkReader) readHeader() error {
 		return err
 	}
 
-	// Verify data checksum
-	data, err := r.buffer.Peek(int(size))
+	// Setup a chunk data buffer so that chunk data can be loaded into it.
+	chunkDataSize := int(size)
+	if chunkDataSize > cap(r.chunkData) {
+		r.chunkData = make([]byte, chunkDataSize)
+	} else {
+		//Reuse existing chunk data buffer if possible
+		r.chunkData = r.chunkData[:chunkDataSize]
+	}
+
+	// To validate checksum of chunk data all the chunk data needs to be loaded into memory at once. Chunk data size is // not bounded to the flush size so peeking chunk data in order to compute checksum may result in bufio's buffer
+	// full error. To circumnavigate this issue load the chunk data into chunk reader's buffer to compute checksum
+	// instead of trying to compute checksum off of fixed size r.buffer by peeking.
+	// See https://github.com/m3db/m3/pull/2148 for details.
+	_, err = io.ReadFull(r.buffer, r.chunkData)
```

Comment:
Yes. This has different semantics. I am loading all the chunk data in chunkData. I have made changes to other places in this code so they read data off chunkData instead of r.buffer

Reply:
Note that I have changed the way we read from buffer because we are not peeking anymore [here](https://github.com/m3db/m3/pull/2277/files#diff-907ca70d2efdead40a453b9631a7ebc3R123) and [here](https://github.com/m3db/m3/pull/2277/files#diff-907ca70d2efdead40a453b9631a7ebc3R146)

Reply:
Whoops sorry, I meant to delete this one haha--explanation makes sense, thanks!